partiql-logical-planner/src/typer.rs

```diff
@@ -0,0 +1,917 @@
+                    StructField::new("id", int!()),
+                    StructField::new("name", str!()),
+                ],
+            ),
+            vec![
+                StructField::new("id", int!()),
+                StructField::new("name", str!()),
+                StructField::new("age", missing!()),
+            ],
+        )
+        .expect("Type");
+
+        // Open Schema with `Strict` typing mode and `age` in nested attribute.
+        let details_fields = struct_fields![("age", int!())];
+        let details = r#struct![BTreeSet::from([details_fields])];
+
+        assert_query_typing(
+            TypingMode::Strict,
+            "SELECT customers.id, customers.name, customers.details.age FROM customers",
+            create_customer_schema(
+                true,
+                vec![
+                    StructField::new("id", int!()),
+                    StructField::new("name", str!()),
+                    StructField::new("details", details.clone()),
+                ],
+            ),
+            vec![
+                StructField::new("id", int!()),
+                StructField::new("name", str!()),
+                StructField::new("age", int!()),
+            ],
+        )
+        .expect("Type");
+
+        // Open Schema with `Strict` typing mode and `bar` in nested attribute.
+        assert_query_typing(
+            TypingMode::Strict,
+            "SELECT customers.id, customers.name, customers.details.age, customers.details.foo.bar FROM customers",
+            create_customer_schema(true,vec![
+                StructField::new("id", int!()),
+                StructField::new("name", str!()),
+                StructField::new("details", details.clone()),
+            ]),
+            vec![
+                StructField::new("id", int!()),
+                StructField::new("name", str!()),
+                StructField::new("age", int!()),
+                StructField::new("bar", any!()),
+            ],
+        )
+        .expect("Type");
+    }
+
+    #[test]
+    fn simple_sfw_with_alias() {
+        // Open Schema with `Strict` typing mode and `age` in nested attribute.
+        let details_fields = struct_fields![("age", int!())];
+        let details = r#struct![BTreeSet::from([details_fields])];
+
+        assert_query_typing(
+            TypingMode::Strict,
+            "SELECT d.age FROM customers.details AS d",
+            create_customer_schema(
+                true,
+                vec![
+                    StructField::new("id", int!()),
+                    StructField::new("name", str!()),
+                    StructField::new("details", details.clone()),
+                ],
+            ),
+            vec![StructField::new("age", int!())],
+        )
+        .expect("Type");
+
+        assert_query_typing(
+            TypingMode::Strict,
+            "SELECT c.id AS my_id, customers.name AS my_name FROM customers AS c",
+            create_customer_schema(
+                false,
```

Comment:
nit: test could have a comment since this test is different than the comment from L709. This test is w/ closed schema + strict typing mode